django/contrib/postgres/fields/serial.py

```diff
@@ -0,0 +1,77 @@
+from django.db import models
+from django.db.models.expressions import DatabaseDefault, Expression
+from django.utils.translation import gettext_lazy as _
+
+__all__ = ("BigSerialField", "SmallSerialField", "SerialField")
+
+
+class SerialFieldSequence(Expression):
+    allowed_default = True
+
+    def as_sql(self, compiler, connection):
+        table = self.field.model._meta.db_table
+        column = self.field.column
+        return f"nextval(pg_get_serial_sequence('{table}', '{column}'))", []
```

Comment:
This can be an SQL injection vector and will crash if any table or column name includes a quote.

```suggestion
        table = self.field.model._meta.db_table
        column = self.field.column
        return "nextval(pg_get_serial_sequence(%s, %s))", [table, column]
```

I would personally use `Func` instead to avoid this error prone boilerplate

```python
class NextSerialSequence(Func):
    template = "nextval(pg_get_serial_sequence(%(expressions)s))"
```

And override `SerialFieldMixin._db_default_expression` to return

```python
@cached_property
def _db_default_expression(self):
    return NextSerialSequence(
        Value(self.model._meta.db_table),
        Value(self.column),
        output_field=self,
    )
```

Which should also avoid the `db_default` injection in `__init__` and removal in `deconstruct`.

Reply:
Thanks @charettes , fair enough. We should really set `db_default` to something though, overriding `_db_default_expression` is not enough because this variable is checked during migrations.